### DISCUSSION TITLE
Blocking grifting

### DISCUSSION THREAD

**Discussion by balupton** [balupton/filters#6] [General]
Would be nice for a filter to block all grifting on producthunt and twitter.

For product hunt, this would be removing any entry that has "GPT" or "AI" inside it. I tried this rule, but couldn't get it to work:

```
producthunt.com##.layoutMain a[href],div.color-lighter-grey:contains(/[ai|gpt]/gi)
```

For twitter, it would be blocking the "For you" tab and blocking anything with "GPT", "AI", anything with a hashtag, anything with an image, anything with an emoji, anything with "here's how", "recap", "below", "step", "needs", "vc", "yc", "harvard", "zuckerberg", "steve jobs", "thank me", "thanks to", "everyone", "brand", "marketing", "hustle", " u ", " ur ", "sold|sell".

**Comment by Franck-Nein** [comment]
Wouldn't that block any div.color-lighter-grey with either an a, g, i, p or t?
As far as I know, at least on uBlock Origin, contains does not allow you to use OR, but in this particular case 
```
producthunt.com##.layoutMain a[href],div.color-lighter-grey:contains(Artificial Intelligence):upward(3)
```
should work, but I have no idea how to do the Twitter on without making a rule per condition.

**Reply by balupton** [reply]
Interesting, I was going off the syntax from https://github.com/bevry-labs/filters/commit/a291ee754789001bc043154a39eab37f125f7573 from @RalphRacer as I never knew of that syntax and wasn't sure if it was regexes. If it is regex, and say an adguard specific filter, we should document that and use rounded brackets instead of square there. @RalphRacer any advice?

edit. Yeah seems the wrong parens were used in that earlier commit, I'll give this another go in a week or so 
https://adguard.com/kb/general/ad-filtering/create-own-filters/#extended-css-contains

**Reply by RalphRacer** [reply]
Yes, this seems to be an Adguard Feature [1], not sure how that works in uBlock Origin. In the Readme.md only Adguard is mentioned.

`contains()` also accepts regex. The regex should work without the parans, as @balupton mentioned in the edit. See https://regexr.com/7f4gh

[1] https://adguard.com/kb/general/ad-filtering/create-own-filters/#extended-css-contains

edit: Just saw this at [1]

> Compatibility with other pseudo-classes
> 
> Synonyms :-abp-contains() and :has-text() are supported for better compatibility.

**Comment by RalphRacer** [comment]
Good idea to remove all these things.
Version for producthunt that might work:

```
producthunt.com##*:contains(/AI|gpt|GPT/):upward(.flex-row-gap-widescreen-undefined)
```

OT: Right now would love to have a filter, that removes yt videos with Apple Vision pro photoshoped on influencer's face. :)

**Reply by balupton** [reply]
This works great:

```
producthunt.com##*:contains(/ai|gpt|ultimate|end|effortlessly|thrilling|challenge|never|life|2x|3x|10x|crypto|coin|nft/i):upward(.direction-row)
```

Will do twitter some other time.

**Reply by RalphRacer** [reply]
Did remove the ignore case flag, so that it wont remove sections containg eg. expl**ai**n (`ai`)

**Comment by balupton** [comment]
Should this be it's own filter, or be part of the cult filter?

**Reply by RalphRacer** [reply]
Might also fit into _agency_:
> companies whose business is attention. 

Edit: ok - _agency_ seems to whole bussinesses themselve. So _cult_ might better, since it is definately not _sin_ or _vanity_.
